RFC/src/RFC-0121_ConsensusEncoding.md

```diff
@@ -0,0 +1,346 @@
+
+#### Commitment
+
+A commitment is a [RistrettoPublicKey] and so has identical encoding.
+
+#### Schnorr Signature
+
+See the [TLU on Schnorr Signatures](https://tlu.tarilabs.com/cryptography/introduction-schnorr-signatures)
+
+A scnorr signature tuple is `<R, s>` where `R` is a [RistrettoPublicKey] and `s` is a the signature scalar wrapped in [RistrettoPrivateKey].
+
+The encoding is fixed at 64-bytes:
+
+```text
+| 32-byte public key | 32-byte scalar |
+```
+
+#### Commitment Signature
+
+A commitment signature tuple consists of a `<R, u, v>` where `R` is the [Pederson commitment](./Glossary.md#commitment) \\(r_u.G + r_v.H\\)
+for the signature scalars `u` and `v`.
+
+The encoding is fixed at 96-bytes:
+
+```text
+| 32-byte commitment (R) | 32-byte scalar (u) | 32-byte scalar (v) |
+```
+
+#### Example
+
+Given the following data and types:
+
+```javascript
+{
+  // Type: Fixed array of 5 bytes
+  short_id: [1,2,3,4,5],
+  // Type: variable length string
+  name: "Case",
+  // Type: unsigned integer
+  age: 40,
+  // Type: struct
+  details: {
+      // Type: variable length string
+      kind: "Hacker"
+  },
+  // Type: nullable varint
+  dob: null
+}
+```
+
+Encoded (hex) as follows:
+
+| short id |len|  name  | age |len| kind         | null? | dob |
+|----------|---|--------|-----|---|--------------|-------|-----|
+|0102030405|04 |43617365| 28  |05 | 4861636b6572 | 00    |     |
+
+Note that nested structs are flattened and the order must be preserved to allow decoding.
+The `00` null byte is important so that for e.g. the `kind` bytes cannot be manipulated to 
+produce the same encoding as non-null `dob`.
+
+### Block Header 
+[block header]: #block-header "Block header"
+
+The block hash pre-image is constructed by first constructing the merge mining hash. Each encoding is concatenated in order as follows:
+
+1. `version` - `u16` converted to a fixed 2-byte array (little endian) 
+2. `height` - `u64` converted to a fixed 8-byte array (little endian) 
+3. `prev_hash` - fixed 32-bytes 
+4. `timestamp` - UNIX timestamp converted to a fixed 8-byte array (little endian) 
+5. `input_mr` - fixed 32-bytes
+6. `output_mmr_size` - `u64` converted to a fixed 8-byte array (little endian)
+7. `witness_mr` - fixed 32-bytes
+8. `kernel_mr` - fixed 32-bytes
+9. `kernel_mmr_size` - `u64` converted to a fixed 8-byte array (little endian)
+10. `total_kernel_offset` - 32-byte Scalar, see [RistrettoSecretKey]
+11. `total_script_offset` - 32-byte Scalar, see [RistrettoSecretKey]
+
+This pre-image is hashed and block hash is constructed, in order, as follows:
+
+1. `merge_mining_hash` - As above
+2. `pow_algo` - enumeration of types of PoW as a single unsigned byte, where `Monero = 0x00` and `Sha3 = 0x01`
+3. `pow_data` - raw variable bytes (no length varint)
+4. `nonce` - the PoW nonce, `u64` converted to a fixed 8-byte array (little endian)
+
+#### Output Features 
+[Output Features]: #output-metadata-and-features "Output Features"
+
+```rust
```

Comment:
mdbook test is failing, you'll need to `ignore` the code blocks 

```suggestion
```rust, ignore
```